### PR TITLE
WIP - sentinel refactor

### PR DESCRIPTION
**Pull request by bradfrizzell**
Refactored the passing of the sentinel value. No need to have a data attribute on the html element. Should just pass the sentinel as required.

### CODE REVIEW

**Comment by bradfrizzell** [comment]
@lannka

File: src/attributes.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    canonicalUrl = self.document.querySelector(
+        'link[rel="canonical"]').href;
+  } catch(err){
+    console/*OK*/.log("Could not get canonicalUrl");
+  }
+
+  attributes._context = {
+    referrer: self.document.referrer,
+    canonicalUrl,
+    pageViewId: getPageViewId(parentWindow),
```
Comment (lannka) [inline src/attributes.js]: We should get values from `documentInfoForDoc(element)`

Comment (lannka) [inline src/attributes.js]: you can get it from `documentInfoForDoc(parentWindow)`

File: src/attributes.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  const width = element.getAttribute('width');
+  const height = element.getAttribute('height');
+  const attributes = {};
+  attributes.width = getLengthNumeral(width);
+  attributes.height = getLengthNumeral(height);
+  const locationHref = parentWindow.location.href;
+
+  let canonicalUrl;
+  try{
+    canonicalUrl = self.document.querySelector(
```
Comment (lannka) [inline src/attributes.js]: don't use `self`, use `parentWindow` instead.
If the naming `parentWindow` is confusing, i suggest to switch to `win`

**Comment by lannka** [comment]
I saw you duplicated some logic from `3p-iframe.js` to `attributes.js`, do you plan to consolidate?

File: src/attributes.js (hunk 1 above)
Comment (bradfrizzell) [inline src/attributes.js]: ah that makes sense. thanks

File: src/attributes.js (hunk 2 above)
Comment (bradfrizzell) [inline src/attributes.js]: can actually use the documentInfoForDoc here as well

**Comment by bradfrizzell** [comment]
Yes, working on consolidating that now.

File: extensions/amp-ad/0.1/amp-ad-3p-impl.js [hunk 3]
```diff
@@ -233,10 +233,11 @@ export class AmpAd3PImpl extends AMP.BaseElement {
       // incrementLoadingAds().
       this.lifecycleReporter.sendPing('adRequestStart');
       const iframe = getIframe(this.element.ownerDocument.defaultView,
-          this.element, undefined, opt_context);
+                               this.element, undefined, opt_context);
```
Comment (jridgewell) [inline extensions/amp-ad/0.1/amp-ad-3p-impl.js]: spacing is incorrect.

File: extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js [hunk 4]
```diff
@@ -74,32 +80,34 @@ export class AmpAdXOriginIframeHandler {
...
           this.element_.creativeId = info.data.id;
         });
 
     // Install iframe resize API.
-    this.unlisteners_.push(listenFor(this.iframe, 'embed-size',
-        (data, source, origin) => {
-          this.updateSize_(data.height, data.width, source, origin);
-        }, true, true));
+    this.unlisteners_.push(listenFor(this.iframe, this.sentinel, 'embed-size',
+                                     (data, source, origin) => {
```
Comment (jridgewell) [inline extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js]: spacing.

File: extensions/amp-ad/0.1/amp-ad-3p-impl.js [hunk 5]
```diff
@@ -233,10 +233,11 @@ export class AmpAd3PImpl extends AMP.BaseElement {
...
+      const sentinel = JSON.parse(iframe.name).attributes._context.sentinel;
```
Comment (jridgewell) [inline extensions/amp-ad/0.1/amp-ad-3p-impl.js]: This is not a JSON encoded value, it's https://github.com/ampproject/amphtml/pull/6506/files#diff-34e84f9cf5e8134d86bf1fc893ab293aR123

Comment (bradfrizzell) [inline extensions/amp-ad/0.1/amp-ad-3p-impl.js]: sorry, this line should be a part of my other PR

File: extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js (hunk 4 above)
Comment (bradfrizzell) [inline extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js]: done

**Pull request closed without merging**